packages/search-types/src/request/request.model.ts

```diff
@@ -0,0 +1,56 @@
+import { Filter } from '../facet/filter/filter.model';
+import { Sort } from '../sort.model';
+
+/**
+ * Interface for any Request with a `query` parameter.
+ *
+ * @public
+ */
+export interface QueryableRequest {
+  query: string;
+}
+
+/**
+ * Interface for any Request with a `filters` parameter.
+ *
+ * @public
+ */
+export interface FilterableRequest {
+  filters?: Record<string, Filter[]>;
```

Comment:
The `Dictionary` type is in `x-utils` package. `x-utils` has `x-types` as dependency. If I want to use the type here I need to add the `x-utils` as a dependency of `x-types`, and then we will have a dependency cycle.

We have to keep the `x-types` without dependencies to other packages, because is the package that is used by all the rest.